Adding common divine spell animations (1-2)

AdamPGameDev opened the pull request:
1st Rank:
bane - does require an aura effect that's not default to Bane, unsure how to export item files otherwise would have included aura effect item. has allowance for free and patreon versions bless - might be a bit load heavy
infuse-vitality - has an on-cast bolt toward targets and a persistent on the effect 
2nd Rank:
spiritual armament - has a simple version for free, for patreon version has different animations for different damage types. The item:slug: gives a warning but thats the only way I could get it to work because of the selector for the roll type

Commit:
Adding spell animations

1st Rank:
bane - does require an aura effect that's not default to Bane, unsure how to export item files otherwise would have included aura effect item. has allowance for free and patreon versions
bless - might be a bit load heavy
infuse-vitality - has an on-cast bolt toward targets and a persistent on the effect
spiritual armament - has a simple version for free, for patreon version has different animations for different damage types. The item:slug: gives a warning but thats the only way I could get it to work because of the selector for the roll type

Dusk-Lynx commented:
is it just bane that triggers multiple times or is it others? if so I can try to fix it.

I thought the next best ranged anim for a melee weapin in jb2a was also a patreon only (I could have been wrong though I'll go back and check) but when I set that one to ranged, It would stretch the animation and basically go over the whole map at anything longer than ~45 ft so it was set to melee to prevent that. However I update to find a better solution to this

Commit:
Update PR, fixing Spiritual Armament and updated line endings

Fixed the spiritual armament issues, realized the options issue was a holdover from when I was using another spell's code to test my slug call at the top.

Dusk-Lynx commented:
for the new commit, I was able to find a ranged animation in free so used that rather than the melee sword. also realized the issue with stretch you were talking about, was a holdover when I was testing my slug call for the spell and removed it

MrVauxs commented:
> is it just bane that triggers multiple times or is it others? if so I can try to fix it.

Just bane and bless. Their predicates are not mutually exclusive, making enabling persistent animations play both the persistent and temporary effect. How you'd do this is with using `contents` and `default: true` which is basically "if nothing else works, use this, otherwise do not play this."

Commit:
Fixed repeated anims

Used "contents" tag to shorten the scripts for Bless, Bane, and Spiritual Armament

Dusk-Lynx commented:
Hopefully that fixes everything! and thank you for being patient with me learning the JSON tags, now that I know about Contents that will help a lot with getting more spells in

MrVauxs commented on animations/spells/1st/infuse-vitality.json:
```diff
@@ -0,0 +1,33 @@
+			}
+		}
+	],
+	"spell:infuse-vitality": [
```
I'd make this ranged animation part of the `effect`. See the Shield cantrip.

MrVauxs commented on animations/spells/1st/bless.json:
```diff
@@ -0,0 +1,41 @@
+								"persistTokenPrototype": true
+							},
+						"tieToDocuments": true,
+						"belowTokens": true
```
You keep repeating belowTokens multiple times. Also you have two defaults. Lastly, the non-persistent version will trigger only once. See other effects for examples on how to format persistent vs. non-persistent effects and copy that.

MrVauxs commented on animations/spells/1st/bane.json:
As bless comment.

Commit:
Updating for requested changes

Bless: Fixed the repeated belowTokens call, fixed the persistent mode to not be default, referenced Haste and Shield for formatting, along with that made non-persistent mode play at start of turn.

Bane: same as bless

Infuse Vitality: referenced Shield as suggested, also made a persistent and non-persistent mode to match Shield and Bless/Bane

Spiritual Armament: added settings quality predicate to fit the non-persistent formatting

Dusk-Lynx commented:
Referenced Shield and Haste as a basis for performing the requested changes, made it so that Bless Bane and Infuse Vitality's non-persistent modes trigger on turn start as well

Commit:
fix errors

Commit:
It's always the missing syntax

Dusk-Lynx commented:
for the record I have no idea what to do with the Vite errors, it seems like it's marking the keywords being used as errors. I assume there's nothing to be done about that check
